File: Data Structures & Algorithms/time-based-key-value-store/submission-1.py
class TimeMap:
    # So every timestamp has a map. 
    def __init__(self):
        self.store = {}
    
    def set(self, key: str, value: str, timestamp: int) -> None:
        if key not in self.store:
            self.store[key] = []
        self.store[key].append([value, timestamp])

    def get(self, key: str, timestamp: int) -> str:
        result = ""
        values = self.store.get(key, [])

        left = 0
        right = len(values) - 1

        while left <= right:
            mid = (right + left) // 2
            val = values[mid]

            if val[1] <= timestamp:
                result = val[0]
                left = mid + 1
            else:
                right = mid - 1

        return result

File: Data Structures & Algorithms/special-array-i/submission-1.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        for (int i = 0; i < nums.length - 1; i++) {
            final int left = nums[i];
            final int right = nums[i + 1];

            if ((left & 1) != (right & 1)) continue;

            return false;
        }

        return true;
    }
}